inspirehep/dojson/hep/fields/bd90x99x.py

```diff
@@ -56,55 +61,105 @@ def get_valid_pubnotes(pubnotes):
                     raw_references.append(pubnote)
         return valid_pubnotes, raw_references
 
-    def get_value(value):
-        recid = None
-        number = ''
-        year = ''
-        if '0' in value:
-            try:
-                recid = int(value.get('0'))
-            except:
-                pass
-        if 'o' in value:
-            try:
-                number = int(value.get('o'))
-            except:
-                pass
-        if 'y' in value:
-            try:
-                year = int(value.get('y'))
-            except:
-                pass
+    def force_force_list(obj):
+        return list(utils.force_list(obj) or [])
 
+    def get_value(value):
+        # Retrieve fields as described here:
+        # https://twiki.cern.ch/twiki/bin/view/Inspire/DevelopmentRecordMarkup.
+
+        # Reference specific.
+        recid = get_int_value(value, '0')
+        number = get_int_value(value, 'o')
+        misc = force_force_list(value.get('m'))
+        raw_references = force_force_list(value.get('x'))
+        urls = force_force_list(value.get('u'))
+        texkey = force_single_element(value.get('1'))
+
+        # Publication info specific.
+        cnum = force_single_element(value.get('b'))
+        report_nums = force_force_list(value.get('r'))
+        pubnotes = force_force_list(value.get('s'))
+        publisher = force_single_element(value.get('p'))
+        year = get_int_value(value, 'y')
+        if year > 2050 or year < 1000:
+            year = None
         try:
-            isbn = normalize_isbn(value['i'])
-        except (KeyError, ISBNError):
-            isbn = ''
-
-        valid_pubnotes, raw_references = get_valid_pubnotes(utils.force_list(value.get('s')))
-
-        if value.get('x'):
-            raw_references += list(utils.force_list(value.get('x')))
+            isbn = normalize_isbn(force_single_element(value.get('i')))
+        except (TypeError, ISBNError):
+            isbn = None
+        doi_or_hdl = force_single_element(value.get('a'))
+
+        # Author specific.
+        author_names = force_force_list(value.get('h'))
+        editor_names = force_force_list(value.get('e'))
+        collaboration = force_force_list(value.get('c'))
+
+        # Titles.
+        titles = force_force_list(value.get('t'))
+        maintitle = force_single_element(value.get('q'))
+        if maintitle:
+            titles.append(maintitle)
+
+        valid_pubnotes, extra_raw_refs = get_valid_pubnotes(pubnotes)
+        raw_references.extend(extra_raw_refs)
+
+        authors = [{'full_name': name} for name in author_names]
+        authors.extend({'full_name': name, 'role': 'ed.'}
+                       for name in editor_names)
+
+        other_report_numbers = []
+        arxiv_eprints = []
+        for num in report_nums:
+            if RE_VALID_ARXIV_REP_NO.match(num):
```

Comment:
Isn't it better to use all the functions from `idutils` https://github.com/inveniosoftware/idutils/blob/master/idutils/__init__.py instead of duplicating here the regexps?